src/packet/bwe/rate_control.rs

```diff
@@ -95,34 +100,115 @@ impl RateControl {
         self.estimated_bitrate
     }
 
-    fn increase(&mut self, now: Instant) {
-        let since_last_update = ((now - self.last_estimate_update).as_millis() as f64) / 1000.0;
+    pub(super) fn set_is_probing(&mut self, is_probing: bool, now: Instant) {
+        if self.is_probing == is_probing {
+            return;
+        }
+        self.is_probing = is_probing;
 
-        let new_estimate = if self.is_near_convergence() {
+        if !is_probing {
+            self.stoppped_probing_at = Some(now);
+        } else {
+            self.stoppped_probing_at = None;
+        }
+    }
+
+    fn increase(&mut self, observed_bitrate: Bitrate, now: Instant) {
+        let last_estimate_update = match self.last_estimate_update {
+            Some(n) => n,
+            None => {
+                self.last_estimate_update = Some(now);
+                now
+            }
+        };
+
+        if self
+            .averaged_observed_bitrate
+            .upper_range(3.0)
+            .map(|upper| observed_bitrate.as_f64() > upper)
+            .unwrap_or(false)
+        {
+            self.averaged_observed_bitrate.reset();
+        }
+
+        let since_last_update = (now - last_estimate_update);
+        assert!(since_last_update >= Duration::ZERO);
+        let near_convergence = self.is_near_convergence();
+
+        let mut new_estimate = if near_convergence {
             // Additive increase
-            let response_time_ms = self.last_rtt_us.map(|rtt| rtt / 1000.0).unwrap_or(0.0) + 100.0;
-            let alpha = 0.5 * (since_last_update / response_time_ms).min(1.0);
+            crate::packet::bwe::macros::log_rate_control_applied_change!("increase_additive");
+            let response_time =
+                self.last_rtt.unwrap_or(Duration::ZERO) + Duration::from_millis(100);
+
+            let alpha =
+                0.5 * (since_last_update.as_secs_f64() / response_time.as_secs_f64()).min(1.0);
             let expected_packet_size = self.estimated_packet_size();
             self.estimated_bitrate.as_f64() + (alpha * expected_packet_size).max(1000.0)
         } else {
             // Multiplicative increase
-            let eta = MULTIPLICATIVE_INCREASE_COEF.powf(since_last_update.min(1.0));
-            eta * self.estimated_bitrate.as_f64()
+            crate::packet::bwe::macros::log_rate_control_applied_change!("increase_multiplicative");
+            let eta = MULTIPLICATIVE_INCREASE_COEF.powf(since_last_update.as_secs_f64().min(1.0));
+            let increase = ((eta - 1.0) * self.estimated_bitrate.as_f64()).max(1_000.0);
+
+            self.estimated_bitrate.as_f64() + increase
         };
-        let max = self
-            .last_observed_bitrate
-            .map(|r| r.as_f64() * MAX_ESTIMATE_RATIO);
+        let max = observed_bitrate.as_f64() * MAX_ESTIMATE_RATIO;
+
+        // Give ourselves a bit of leeway to establish the new bitrate when we stop probing.
```

Comment:
This shouldn't happen anymore now that we properly generate padding at a rate near the estimate, remove